Slight for loop gas optimization

redsh4de opened the pull request:
- Changed increment operations to use pre-increment instead of post-increment to save an extra operation
- for loop iterator incrementing is moved into an unchecked block to skip overflow checks and save gas - given that `i` has a maximum limit set by the loop, there are no downsides for this approach

redsh4de committed:
Slight for loop gas optimization
Use unchecked and pre-increment for for-loop iterators

mudgen commented on contracts/libraries/LibDiamond.sol:
```diff
@@ -102,7 +103,8 @@ library LibDiamond {
             addFunction(ds, selector, selectorPosition, _facetAddress);
-            selectorPosition++;
+            ++selectorPosition;
+            unchecked { ++selectorIndex; }
```
> for (uint256 selectorIndex; selectorIndex < _functionSelectors.length; selectorIndex++)

`selectorPosition` is now being incremented twice in each iteration of the loop, instead of once.

mudgen commented:
@redsh4de  I appreciate this. I found one bug in the pull request. Please fix and I can merge.  Thanks for your help.

redsh4de committed:
Update LibDiamond.sol - remove redundant selectorIndex increment from for loop

Remove selectorIndex++ from the for loop, as incrementing is being done in an unchecked block already

redsh4de commented:
That should take care of it! @mudgen
